base/multi.jl

```diff
@@ -962,6 +928,21 @@ function create_message_handler_loop(r_stream::AsyncStream, w_stream::AsyncStrea
     end
 end
 
+function connect_to_peer(manager::ClusterManager, rpid::Integer, wconfig::WorkerConfig)
+    try
+        (r_s, w_s) = connect(manager, rpid, wconfig)
+        w = Worker(rpid, r_s, w_s, manager, wconfig)
+        process_messages(w.r_stream, w.w_stream)
+        send_msg_now(w, :identify_socket, myid())
+
+        # test connectivity with an echo
+        assert(:ok == remotecall_fetch(rpid, ()->:ok))
```

Comment:
wouldn't it be better to throw do the check and throw a better error here?  Otherwise it will be caught and the printed error will be 
```julia
Error: AssertionError("") connecting to peer. Exiting
```